core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/TransformCorrelatedScalarSubquery.java

```diff
@@ -101,6 +102,7 @@ public Pattern<CorrelatedJoinNode> getPattern()
     @Override
     public Result apply(CorrelatedJoinNode correlatedJoinNode, Captures captures, Context context)
     {
+        checkArgument(correlatedJoinNode.getType() == INNER || correlatedJoinNode.getType() == LEFT, "unexpected correlated join type: " + correlatedJoinNode.getType());
```

Comment:
add a comment explaining why only INNER and LEFT are supposed to show up here and make it a `verify` (this is assertion-like check)

why isn't it a part of the PATTERN?